src/seedu/addressbook/parser/Parser.java

```diff
@@ -238,16 +241,36 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
      * @return the prepared command
      */
     private Command prepareFind(String args) {
-        final Matcher matcher = KEYWORDS_ARGS_FORMAT.matcher(args.trim());
-        if (!matcher.matches()) {
+        final Matcher matcherTagsOnly = TAGS_ARGS_FORMAT.matcher(args.trim());
+        final Matcher matcherNamesOnly = NAMES_ARGS_FORMAT.matcher(args.trim());
+        Matcher matcher;
+        boolean isFindTag = false;
+
+        if (!matcherNamesOnly.matches() && !matcherTagsOnly.matches()) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                     FindCommand.MESSAGE_USAGE));
         }
+        if (matcherTagsOnly.matches()) {
+            matcher = matcherTagsOnly;
+            isFindTag = true;
+        } else {
+            matcher = matcherNamesOnly;
+        }
+
+
+        //  tagArguments and nameArguments delimited by whitespace
+        String[] tagArguments, nameArguments;
+        if (isFindTag) {
+            tagArguments = matcher.group("tagArguments").split("\\s+");
+            nameArguments = new String[0];
+        } else {
+            nameArguments = matcher.group("nameArguments").split("\\s+");
+            tagArguments = new String[0];
+        }
 
-        // keywords delimited by whitespace
-        final String[] keywords = matcher.group("keywords").split("\\s+");
-        final Set<String> keywordSet = new HashSet<>(Arrays.asList(keywords));
-        return new FindCommand(keywordSet);
+        final Set<String> nameSet = new HashSet<>(Arrays.asList(nameArguments));
+        final Set<String> tagSet = new HashSet<>(Arrays.asList(tagArguments));
+        return new FindCommand(nameSet, tagSet);
```

Comment:
So for this maybe better to just have one Set containing either names or tags, then pass it with a boolean denoting which one it is, rather than make an empty object?